packages/frontend/src/creation-form/index.tsx

```diff
@@ -1,12 +1,124 @@
 import "../global.css";
 import "@carrot-kpi/ui/styles.css";
 
-import { type ReactElement } from "react";
-import { type OracleRemoteCreationFormProps } from "@carrot-kpi/react";
+import { type ReactElement, useState, useEffect } from "react";
+import {
+    useDecentralizedStorageUploader,
+    type OracleRemoteCreationFormProps,
+    type OracleInitializationBundleGetter,
+} from "@carrot-kpi/react";
+import { Select, Typography } from "@carrot-kpi/ui";
+import { type MetricOption, type Specification, type State } from "../types";
+import { DEFILLAMA_ANSWERER_URL, METRICS } from "../commons";
+import { PayloadForm } from "./payload-form";
+import { encodeAbiParameters } from "viem";
 
-interface State {}
+export const Component = ({
+    state,
+    onChange,
+    t,
+    kpiToken,
+}: OracleRemoteCreationFormProps<State>): ReactElement => {
+    const uploadToIpfs = useDecentralizedStorageUploader();
 
-export const Component =
-    ({}: OracleRemoteCreationFormProps<State>): ReactElement => {
-        return <></>;
-    };
+    const [metric, setMetric] = useState<MetricOption | undefined>(
+        METRICS.find((metric) => metric.value === state.specification?.metric),
+    );
+    const [payload, setPayload] = useState<
+        Specification["payload"] | undefined
+    >(state.specification?.payload);
+
+    // this effect reacts to any change in internal state, firing an
+    // onChange event to the creation form when necessary.
+    // onChange will also receive an initialization bundle getter function
+    // when data is valid.
+    useEffect(() => {
+        let cancelled = false;
+        const validateAndChange = async () => {
+            const specification = {
+                metric: metric?.value,
+                payload,
+            };
+            const newState: State = { specification };
+
```

Comment:
I think we should add a condition here, like the following:

```ts
if (
    !!specification.metric ||
    !!specification.payload?.protocol ||
    !!specification.payload?.timestamp
)
    return;
```

With this, we'd avoid making HTTP calls to the `/validations` that we know would fail, since the payload is incomplete.

Reply:
Nice catch, I've added the check in a27d86bea05fac63494443db2a86b72af5f16e6f